docs/src/main/asciidoc/kotlin.adoc

```diff
@@ -0,0 +1,26 @@
+== Kotlin Support
+
+The latest version of the Spring Framework provides first-class support for Kotlin.
+For Kotlin users of Spring, the Spring Cloud GCP libraries work out-of-the-box and are fully interoperable with Kotlin applications.
+
+For more information on building a Spring Boot application in Kotlin, please consult the https://docs.spring.io/spring/docs/current/spring-framework-reference/languages.html#kotlin[Spring Kotlin documentation].
+
+=== Prerequisites
+
+Ensure that your Kotlin Spring Boot application is properly set up.
+Based on your build system, you will need to include the correct Kotlin build plugin in your project:
+
+* https://kotlinlang.org/docs/reference/using-maven.html[Kotlin Maven Plugin]
+* https://kotlinlang.org/docs/reference/using-gradle.html[Kotlin Gradle Plugin]
+
+Depending on your application's needs, you may need to augment your build configuration with compiler plugins:
+
+* https://kotlinlang.org/docs/reference/compiler-plugins.html#spring-support[Spring Kotlin Plugin]: Makes your Spring configuration classes/members non-final for convenience.
+* https://kotlinlang.org/docs/reference/compiler-plugins.html#jpa-support[Kotlin Spring JPA Plugin]: Enables using JPA in Kotlin Spring applications.
```

Comment:
Should we standardize "Spring Kotlin" vs "Kotlin Spring" for consistency? Grammatically, I think, it makes no difference.

Reply:
Sounds good.

@elefeint - Moved to Kotlin Spring for consistency; it appears the spring docs refer to the plugin as "kotlin-spring" so I will standardize around that.

@artembilan - Renamed to Kotlin JPA plugin to be more accurate.